automation/label-approved-veps.py

```diff
@@ -0,0 +1,96 @@
+import os
+import re
+import requests
+
+GITHUB_TOKEN = os.environ["GITHUB_TOKEN"]
+KUBEVIRT_REPO = os.environ.get("GITHUB_REPOSITORY", "kubevirt/kubevirt")
+PR_NUMBER = os.environ["PR_NUMBER"]
+HEADERS = {"Authorization": f"token {GITHUB_TOKEN}", "Accept": "application/vnd.github.v3+json"}
+
+def get_pr_details():
+    #Fetch the kubevirt/kuebvirt PR body
+    url = f"https://api.github.com/repos/{KUBEVIRT_REPO}/pulls/{PR_NUMBER}"
+    response = requests.get(url, headers=HEADERS)
+    response.raise_for_status()
+    return response.json()["body"]
+
+def extract_enhancements_references(pr_body):
+    #Get the enhancements reference numbers from the PR.
+    pattern = r"(?:https://github.com/)?kubevirt/enhancements/(?:issues|pull)/(\d+)|(?:kubevirt/)?enhancements#(\d+)"
+    matches = re.findall(pattern, pr_body)
+
+    # Extract the first non-empty group from each match
+    ref_numbers = [group[0] or group[1] for group in matches]
+    # there can be duplicates
+    return list(set(ref_numbers))
+
+def is_merged_pr(ref_number):
+    url = f"https://api.github.com/repos/kubevirt/enhancements/pulls/{ref_number}"
+    response = requests.get(url, headers=HEADERS)
+    if response.status_code == 200:
+        pr_data = response.json()
+        return pr_data.get("merged_at") is not None
+    return False
+
+def find_related_merged_prs(issue_number):
+    url = "https://api.github.com/repos/kubevirt/enhancements/pulls?state=closed&per_page=100"
```

Comment:
```suggestion
    url = "https://api.github.com/search/issues?q=repo:kubevirt/enhancements+is:pr+is:merged&per_page=100"
```
That would reduce the list slightly, but we should still reach 100 somewhat soon, so we probably need to support multiple pages...

Reply:
ok... added pagination ;)